src/persist-client/src/internal/gc.rs

```diff
@@ -360,6 +368,41 @@ where
             req.shard_id, req.new_seqno_since
         );
 
+        // Finally, apply the remaining diffs and calculate the live / current
+        // set of diffs and blobs for metrics.
+        while let Some(state) = states.next() {
+            live_diffs += 1;
+            state.collections.trace.map_batches(|b| {
+                for part in b.parts.iter() {
+                    live_batch_blobs.insert(part.key.to_owned());
+                }
+            });
+        }
+
+        let state = states.state();
+
+        let live_batch_blobs: i64 = live_batch_blobs
+            .len()
+            .try_into()
+            .expect("live blob count fits into i64");
+
+        let current_batch_blobs: i64 = {
+            let mut current_batch_blobs = HashSet::new();
+            state.collections.trace.map_batches(|b| {
+                for part in b.parts.iter() {
+                    current_batch_blobs.insert(part.key.to_owned());
```

Comment:
it's safe to assume each part key in current state is unique and just count the number of parts (to save us the HashSet)